### PR TITLE
add fuzzers

### PR DESCRIPTION
**Pull request by dafifynn**
This PR creates different basic fuzzer examples, they are going to be listed down.

* Differential fuzzer of different deflate implementations using libflate and inflate

### CODE REVIEW

**Comment by dafifynn** [comment]
this csv-parse fuzzer fund a crash in csv-parse library 
Captura de pantalla 2022-11-22 a la(s) 19 40 07

[here](https://github.com/paul-aspinall/csv-parser/blob/master/src/lib.rs) at line 73 there´s an unwrap of an Error, to recreate this run `cargo hfuzz run-debug csv-parser ../csv-parser/hfuzz_workspace/csv-parser/SIGABRT.PC.fffff7e09eac.STACK.1b3f786f57.CODE.-6.ADDR.0.INSTR.adrp_x0,_0x000000000013c000.fuzz `

**Comment by dafifynn** [comment]
this fuzzer was created with a corpora collection of .asn1 files and  crashed when it couldn´t parse a very short data file, still a work in progress as my intention is to test decoding of der files
Captura de pantalla 2022-11-23 a la(s) 21 48 58

**Review by igaray — CHANGES_REQUESTED** [review]
@dafifynn Is it possible that the `asn1/hfuzz_target` cache directory was not meant to be included? In addition, could you remove the .DS_store files?

**Comment by daphneherlambda** [comment]
> @dafifynn Is it possible that the `asn1/hfuzz_target` cache directory was not meant to be included? In addition, could you remove the .DS_store files?

yes!!! I miss delete the extra files in asn1 fuzzer, also the .DS_store ones, they´re gone now

**Review by igaray — APPROVED** [review]

File: image-tiff/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#[macro_use]
+extern crate honggfuzz;
+extern crate tiff;
+
+// to run this fuzzer use honggfuzz to run image-tiff
+use std::io::Cursor;
+
+fn main() {
+    loop {
+        fuzz!(|data: &[u8]| {
```
Comment (igaray) [inline image-tiff/src/main.rs:10]: Shoudn't this value be used in the test body?

**Review by igaray — CHANGES_REQUESTED** [review]

File: image-tiff/src/main.rs (hunk 1 above)
Comment (daphneherlambda) [reply image-tiff/src/main.rs:10]: done!! now the cursor is created from the data

**Review by igaray — APPROVED** [review]

**Review by Jrigada — APPROVED** [review]
LGTM!

**Pull request merged**